DependencyInjection/DoctrineExtension.php

```diff
@@ -279,8 +279,16 @@ protected function getConnectionOptions($connection)
                     continue;
                 }
                 $options['master'][$key] = $value;
+                if (!isset($options['master']['driver'])) {
+                    $options['master']['driver'] = $options['driver'];
```

Comment:
Originally it was for the sharding configuration: a connection needs a `driver` or a `driverOptions` option. I removed it from _master/slave_ configuration as my PR only concerns sharding.